EpikCord/client/commands.py

```diff
@@ -49,17 +52,455 @@ def to_dict(self):
     ...
+
+class ClientUserCommand(ClientContextMenuCommand):
+    def __init__(
+        self,
+        name: str,
+        callback: AsyncFunction,
+        *,
+        guild_ids: List[int] = [],
+        name_localizations: List[Localization] = [],
+        guild_only: Optional[bool] = None,
+        default_member_permissions: Optional[Permissions] = None,
+        nsfw: bool = False,
+    ):
+        super().__init__(
+            name,
+            callback,
+            guild_ids=guild_ids,
+            name_localizations=name_localizations,
+            type=ApplicationCommandType.USER,
+            guild_only=guild_only,
+            default_member_permissions=default_member_permissions,
+            nsfw=nsfw,
+        )
+
+class ClientMessageCommand(ClientContextMenuCommand):
+    def __init__(
+        self,
+        name: str,
+        callback: AsyncFunction,
+        *,
+        guild_ids: List[int] = [],
+        name_localizations: List[Localization] = [],
+        guild_only: Optional[bool] = None,
+        default_member_permissions: Optional[Permissions] = None,
+        nsfw: bool = False,
+    ):
+        super().__init__(
+            name,
+            callback,
+            guild_ids=guild_ids,
+            name_localizations=name_localizations,
+            type=ApplicationCommandType.MESSAGE,
+            guild_only=guild_only,
+            default_member_permissions=default_member_permissions,
+            nsfw=nsfw,
+        )
+
+class ApplicationCommandOptionChoice:
+    def __init__(self, name: str, value: str, *, name_localizations: List[Localization] = []):
+        self.name = name
+        self.value = value
+        self.name_localizations = name_localizations
+
+    def to_dict(self):
+        payload: Dict[str, Union[str, Dict[str, str]]] = {
+            "name": self.name,
+            "value": self.value,
+        }
+
+        if name_localizations := localization_list_to_dict(
+            self.name_localizations
+        ):
+            payload["name_localizations"] = name_localizations
+
+        return payload
+
+
+class BaseApplicationCommandOption:
+    def __init__(
+        self,
+        name: str,
+        description: str,
+        *,
+        name_localizations: List[Localization] = [],
+        description_localizations: List[Localization] = [],
+        required: bool = False,
+        type: ApplicationCommandOptionType,
+    ):
+        self.name = name
+        self.description = description
+        self.name_localizations = name_localizations
+        self.description_localizations = description_localizations
+        self.required = required
+        self.type = type
+
+    def to_dict(self):
+        payload = {
+            "name": self.name,
+            "description": self.description,
+            "required": self.required,
+            "type": self.type.value,
+        }
+
+        if name_localizations := localization_list_to_dict(
```

Comment:
We dont need name as long for a single function.

Long name have importance when used globally, as they provide better context.
In a small scope, having long names for such duration make the reading more tedious.

Reply:
What other name would you suggest?

Reply:
Like just `name_loc`

Reply:
So the variable name is the issue, and not the function name?

Reply:
Well would have named the function `get_localization_dict` or something like that but, not disturbing as it is available i a wide range